### ISSUE TITLE
iter_row_groups raises KeyError when dots in column names and data is not primitive

### ISSUE THREAD

**Issue by adrienDog** [dask/fastparquet#935]
**TL;DR**
Column names with dots are badly handled when the column data is not a primitive, e.g. `List[str]`.

-> It wrongly tries to find a key named `foo` if the column is `"foo.with.lists": [[], [], []],`

**Describe the issue**:
Here is a unit test that shows the difference of behaviour and the reproduced bug

```python
import fastparquet
import polars

import pytest


class TestReproduceGroupKeyErrorForNamesWithDots:
    def test_valid_data(self, tmp_path: Path) -> None:
        """Test that column names with dots are not causing issues when cell contents are primitives."""
        df = polars.DataFrame(
            {
                "foo.with.strings": ["hey", "there", None],
                "foo.with.ints": [1, 2, 3],
            }
        )
        df.write_parquet(tmp_path / "output.parquet")

        file = fastparquet.ParquetFile(tmp_path / "output.parquet")

        d = dict(enumerate(file.iter_row_groups(), start=1))
        assert len(d) == 1

    def test_reproduce_error(self, tmp_path: Path) -> None:
        """Test that column names with dots are causing issues when cell contents are not primitives."""
        df = polars.DataFrame(
            {
                "foo.with.strings": ["hey", "there", None],
                "foo.with.ints": [1, 2, 3],
                "foo.with.lists": [[], [], []],
            }
        )
        df.write_parquet(tmp_path / "output.parquet")

        file = fastparquet.ParquetFile(tmp_path / "output.parquet")

        with pytest.raises(KeyError): # for key 'foo'
            d = dict(enumerate(file.iter_row_groups(), start=1))
            assert len(d) == 1
```

**Anything else we need to know?**:
This is heavily used by frictionless-py [here](https://github.com/frictionlessdata/frictionless-py/blob/main/frictionless/formats/parquet/parser.py#L36)

**Environment**:
- Python version: Python 3.9.18
- Operating System: macos
- Install method (conda, pip, source): rye (pip)

**Comment by martindurant** [comment]
Does the same data load via to_pandas, versus iterating?

**Comment by adrien-owkin** [comment]
Hi @martindurant , thank you for the very fast reply ⚡ 

Same error with file.to_pandas() yes

```python
    def test_load_problematic_data_with_to_pandas(self, tmp_path: Path) -> None:
        """Test that column names with dots are causing issues when cell contents are not primitives."""
        df = polars.DataFrame(
            {
                "target": ["ENSG00000154175", "ENSG00000154175", "ENSG00000154175"],
                "gene_name": ["A1BG", "A1BG", "A1BG"],
                "indication": ["indication", "indication", "indication"],
                "foo.with.strings": ["hey", "there", None],
                "foo.with.ints": [1, 2, 3],
                "foo.with.lists": [[], [], []],
            }
        )
        df.write_parquet(tmp_path / "output_table.parquet")

        file = fastparquet.ParquetFile(tmp_path / "output_table.parquet")

        with pytest.raises(KeyError):
            pandas_df  = file.to_pandas() # this raises
            assert pandas_df.shape == (3, 6)
```

the error (when removing the pytest.raises)
```
self = <Parquet Schema with 9 entries>, name = ['foo']

    def schema_element(self, name):
        """Get the schema element with the given name or path"""
        root = self.root
        if isinstance(name, str):
            name = name.split('.')
        for part in name:
>           root = root["children"][part]
E           KeyError: 'foo'
```

**Comment by adrien-owkin** [comment]
After some debugging, it seems that the argument passed to `schema_element` is always a list e.g. `['foo.with.strings']` except for columns with non primitive types where the name is simply a string hence is split by `.`

**Comment by martindurant** [comment]
I'll look into it. Certainly, it is a sloppy convention that "columns" are specified both by ["path", "subpath"] and "path.subpath", where the components  could in theory contain "." characters. 

Interesting that you are using polars to generate the data. Can I ask what use fastparquet is for you?

**Comment by adrien-owkin** [comment]
good point ^^ 

im using `frictionless-py` to validate a datapackage (see datapackage.org specs).

and the frictionless-py plugin for parquet uses fastparquet. and, the dots in column names are not in my control atm, defined upstream.

as a workaround, i might override the parquet plugin using polars since it is in my dependencies.

**Comment by martindurant** [comment]
This actually does load on `main`, almost **IF** the array-of-lists column contains some data. Otherwise it hits a point of trying to decompress a zero-length buffer.

**Comment by adrien-owkin** [comment]
you mean this works?
```python
        df = polars.DataFrame(
            {
                "foo.with.strings": ["hey", "there", None],
                "foo.with.ints": [1, 2, 3],
                "foo.with.lists": [[], [], []],
            }
        )
        df.write_parquet(tmp_path / "output_table.parquet")

        file = fastparquet.ParquetFile(tmp_path / "output_table.parquet")

        # no error here 👇 
        pandas_df  = file.to_pandas()
```

with the unit test i could also reproduce with non empty lists
```python
"foo.with.lists": [["a"], ["b"], ["c"]],
```

**Comment by martindurant** [comment]
Correct, if you include data in foo.with.lists (I used [[0], [0], [0]]), it loads. I don't immediately see a PR that might be responsible for that, but many deps have also updated a lot in the meantime, particularly pandas.
